pkg/controller/pxcbackup/controller.go

```diff
@@ -272,7 +272,18 @@ func (r *ReconcilePerconaXtraDBClusterBackup) tryRunS3BackupFinalizerJob(cr *api
 	}
 
 	if cr.Status.S3 == nil || cr.Status.Destination == "" || !strings.HasPrefix(cr.Status.Destination, "s3://") {
+		finalizersBefore := len(cr.GetFinalizers())
 		removeS3Finalizer(cr)
+
+		if finalizersBefore == len(cr.GetFinalizers()) {
+			return nil
+		}
+
+		if err := r.client.Update(context.TODO(), cr); err != nil {
```

Comment:
I am not sure maybe `r.client.Update` can be part of `removeS3Finalizer` function

Reply:
in the function below there are 
https://github.com/percona/percona-xtradb-cluster-operator/blob/daead24b10147355eaeca27e1c7e6b39e61cffa2/pkg/controller/pxcbackup/controller.go#L291
and
https://github.com/percona/percona-xtradb-cluster-operator/blob/daead24b10147355eaeca27e1c7e6b39e61cffa2/pkg/controller/pxcbackup/controller.go#L297

also it creates goroutine
https://github.com/percona/percona-xtradb-cluster-operator/blob/daead24b10147355eaeca27e1c7e6b39e61cffa2/pkg/controller/pxcbackup/controller.go#L291
that call the function
https://github.com/percona/percona-xtradb-cluster-operator/blob/daead24b10147355eaeca27e1c7e6b39e61cffa2/pkg/controller/pxcbackup/controller.go#L372

Reply:
yes, I'd prefer the update in `removeS3Finalizer`

Reply:
moved to `removeS3Finalizer`. optimised memory usage